refactor: BlockExtensions

MukjepScarlet opened the pull request:
- Clean up duplicate code
- Improve readability and some performance

Core changes:
- Use Sequence API to create a lazy BlockPos collection because we usually apply many operations like `map`/`filter` on it
- Use Region API to iterate with BlockPos.Mutable, which reduces memory usage drastically

MukjepScarlet committed:
refactor: BlockExtensions

- Clean up duplicate code
- Improve readability and some performance

MukjepScarlet committed:
detekt fix

ccetl commented on src/main/kotlin/net/ccbluex/liquidbounce/utils/render/placement/PlacementRenderHandler.kt:
```diff
@@ -126,16 +126,16 @@ class PlacementRenderHandler(private val placementRenderer: PlacementRenderer, v
-        scanBlocksInCuboid(2, pos) { // TODO in theory a one block radius should be enough
-            inList.computeIfPresent(it) { _, value ->
-                Triple(value.first, getCullData(it), value.third)
-            }?.let { return@scanBlocksInCuboid false }
```
Why did you remove the early return here?

MukjepScarlet replied:
The previous code always returns false, and the `scanBlocksInCuboid` will stop if it returns true, so I think it's unnecessary

ccetl replied:
It prevented an unnecessary operation tho.

MukjepScarlet replied:
It just prevents the current iteration. I'll add this back. (`return@forEach`)

ccetl replied:
The second return was indeed useless, only the first was useful.

MukjepScarlet committed:
add `continue` (return@forEach) back

MukjepScarlet commented:
@ccetl I think you can use more specific type names with these containers in `PlacementRenderHandler.kt`

MukjepScarlet committed:
Merge branch 'nextgen' into refactor/block

MukjepScarlet committed:
fix: wrong Region intersection check

It used to be `contains` literally

MukjepScarlet committed:
# Conflicts:
#	src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/ModuleNuker.kt

MukjepScarlet committed:
fit new nuker

MukjepScarlet committed:
Merge branch 'nextgen' into refactor/block

MukjepScarlet committed:
Update SphereNukerArea.kt

Sequence -> List

MukjepScarlet committed:
feat: add RelativeToPlayer option (Nuker)

MukjepScarlet committed:
detekt
